toolium/selenoid.py

```diff
@@ -0,0 +1,199 @@
+                self.password = driver_wrapper.config.get('Server', 'password')
+                self.ggr_host = driver_wrapper.config.get('Server', 'host')
+                self.ggr_port = driver_wrapper.config.get('Server', 'port')
+            except Exception as e:
+                driver_wrapper.logger.warn("verify 'username' and 'password', 'host' and 'port' properties below 'Server' in the toolium.cfg: \n  %s" % e)
+            driver_wrapper.logger.info("Selenoid host info: \n %s" % self.get_selenoid_info())
+
+    def __download_file(self, url, path_file, timeout):
+        """
+        download a file from the server using a request with retries policy
+        :param url: server url to request
+        :param path_file: path and file where to download
+        :param timeout: threshold until the video file is downloaded
+        :return boolean
+        """
+        status_code = 0
+        init_time = time.time()
+        # retries policy
+        while status_code != STATUS_OK and time.time() - init_time < float(timeout):
+            body = requests.get(url)
+            status_code = body.status_code
+        took = time.time() - init_time    # time used to download the file
+        # create the folders and store the file downloaded
+        if status_code == STATUS_OK:
+            path, name = os.path.split(path_file)
+            if not os.path.exists(path):
+                os.makedirs(path)
+            try:
+                fp = open(path_file, 'wb')
+                fp.write(body.content)
+                fp.close()
+                self.driver_wrapper.logger.info('the %s file has been downloaded successfully and took %d seconds' % (path_file, took))
+                return True
+            except IOError as e:
+                self.driver_wrapper.logger.warn('the %s file has a problem; \n %s' % (path_file, e))
+        else:
+            self.driver_wrapper.logger.warn('the file to download does not exist in the server after %s s (timeout).' % timeout)
+        return False
+
+    def __remove_file(self, url):
+        """
+        remove a file in the Selenoid node
+        """
+        requests.delete(url)
+
+    def get_selenoid_info(self):
+        """
+        retrieve the current selenoid host info. Used only in hub mode (GGR)
+        request: http://<username>:<password>@<ggr_host>:<ggr_port>/host/<ggr_session_id>
+        :return: dict
+        """
+        if self.hub:
+            url = 'http://{0}:{1}@{2}:{3}/host/{4}'.\
+                format(self.username, self.password, self.ggr_host, self.ggr_port, self.session_id)
+            try:
+                response = requests.get(url)
+            except Exception as e:
+                self.driver_wrapper.logger.warn("the GGR request to get the node host is failed: \n  %s" % e)
+                return None
+            return response.json()
+        return None
+
+    def download_video_if_the_scenario_fails(self, scenario_name, timeout=10):
```

Comment:
remove _if_the_scenario_fails

Reply:
The  video is always removed, if the scenario fails or not, see below:
```
  self.__remove_file(server_url)
```

Reply:
Fixed in 2bcebd5